source/main/gui/panels/GUI_GameMainMenu.cpp

```diff
@@ -84,58 +108,97 @@ void GameMainMenu::DrawMenuPanel()
     else
     {
         const float btn_height = ImGui::GetTextLineHeight() + (BUTTON_PADDING.y * 2);
-        const float window_height = (6*btn_height) + (5*ImGui::GetStyle().ItemSpacing.y) + (2*ImGui::GetStyle().WindowPadding.y); // 5 buttons + titlebar; 2x spacing around separator
+        const float window_height = (m_num_buttons*btn_height) + (m_num_buttons*ImGui::GetStyle().ItemSpacing.y) + (2*ImGui::GetStyle().WindowPadding.y); // buttons + titlebar; 2x spacing around separator
         const float margin = display_size.y / 15.f;
         const float top = display_size.y - window_height - margin;
         ImGui::SetNextWindowPos(ImVec2(margin, top));
     }
     ImGui::SetNextWindowContentWidth(WINDOW_WIDTH);
     int flags = ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_AlwaysAutoResize;
-    if (ImGui::Begin(_LC("MainMenu", "Main menu"), nullptr, static_cast<ImGuiWindowFlags_>(flags)))
+    if (ImGui::Begin(_LC("MainMenu", title), nullptr, static_cast<ImGuiWindowFlags_>(flags)))
     {
         int button_index = 0;
         ImVec2 btn_size(WINDOW_WIDTH, 0.f);
-
-        if (HighlightButton(_LC("MainMenu", "Single player"), btn_size, button_index++))
+        if (App::app_state->GetEnum<AppState>() == AppState::MAIN_MENU)
         {
-            this->SetVisible(false);
-            if (App::diag_preset_terrain->GetStr().empty())
-            {
-                RoR::Message m(MSG_GUI_OPEN_SELECTOR_REQUESTED);
-                m.payload = reinterpret_cast<void*>(new LoaderType(LT_Terrain));
-                App::GetGameContext()->PushMessage(m);
-            }
-            else
+            if (HighlightButton(_LC("MainMenu", "Single player"), btn_size, button_index++))
```

Comment:
what about keeping this button visible even in simulation? It will open selector for new map, and if cancelled, user stays in current simulation. You'll have to adjust the Selector/OkButton code a bit (to also send UNLOAD and DISCONNECT), but I think it's the way to go.